src/app.js

```diff
@@ -381,24 +355,90 @@ function setupWebSocketsServer(wsServer) {
 
 /**
  *
+ * @param {*} request
+ * @param {*} response
  */
-function run() {
-    logger.info('[App] Initializing: %s; version: %s; env: %s ...', appName, appVersion, getEnvName());
-    let server;
+function serverHandler(request, response) {
+    switch (request.url) {
+    case '/healthcheck':
+        response.writeHead(200);
+        response.end();
+        break;
+    case '/bindcheck':
+        logger.info('Accessing bind check!');
+        response.writeHead(200);
+        response.end();
+        break;
+    default:
+        response.writeHead(404);
+        response.end();
+    }
+}
 
-    setupWorkDirectory();
+/**
+ * In case one wants to run the server locally, https is required, as browsers normally won't allow non
+ * secure web sockets on a https domain, so something like the bello
+ * server instead of http.
+ *
+ * @param {number} port
+ */
+function setupHttpsServer(port) {
+    const { keyPath, certPath } = config.get('server');
 
-    if (config.get('server').useHTTPS) {
-        server = setupHttpsServer(config.get('server').port);
-    } else {
-        server = setupHttpServer(config.get('server').port);
+    if (!(keyPath && certPath)) {
+        throw new Error('[App] Please provide certificates for the https server!');
+    }
+
+    const options = {
+        key: fs.readFileSync(keyPath),
+        cert: fs.readFileSync(certPath)
+    };
+
+    return https.createServer(options, serverHandler).listen(port);
+}
+
+/**
+ *
+ * @param {*} port
+ */
+function setupHttpServer(port) {
+    return http.createServer(serverHandler).listen(port);
+}
+
+
+/**
+ *
+ */
+function setupWebServer() {
+    const { useHTTPS, port } = config.get('server');
+
+    if (!port) {
+        throw new Error('[App] Please provide a server port!');
     }
 
-    if (config.get('server').metrics) {
-        setupMetricsServer(config.get('server').metrics);
+    let server;
+
+    if (useHTTPS) {
+        server = setupHttpsServer(port);
+    } else {
+        server = setupHttpServer(port);
     }
 
     setupWebSocketsServer(server);
+}
+
+/**
+ *
+ */
+function run() {
+    logger.info('[App] Initializing: %s; version: %s; env: %s ...', appName, appVersion, getEnvName());
+
+    setupWorkDirectory();
+    setupDumpStorage();
+    setupFeaturesPublisher();
+    setupAmplitudeConnector();
+    setupWebServer();
+    setupMetricsServer();
```

Comment:
Just a little nit: it might be best to start the actual web server last, so that everything is ready once traffic gets processed.

Reply:
makes sense, will do that.